modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java

```diff
@@ -415,14 +415,26 @@ private void exec() {
 
                 case InstructionCodes.LENGTHOF:
                     i = operands[0];
-                    j = operands[1];
-                    if (sf.refRegs[i] == null) {
-                        handleNullRefError();
+                    cpIndex = operands[1];
+                    j = operands[2];
+
+                    typeRefCPEntry = (TypeRefCPEntry) constPool[cpIndex];
+                    int typeTag = typeRefCPEntry.getType().getTag();
+                    if (typeTag == TypeTags.STRING_TAG) {
+                        sf.longRegs[j] = sf.stringRegs[i].length();
```

Comment:
this can return a null pointer exception.

Reply:
Fixed